### ISSUE TITLE
Load event not working

### ISSUE THREAD

**Issue by frabanca** [highcharts/highcharts-angular#115]
Hi, 

I'm having an issue with highcharts-angular while try to hook load event. 

**My versions**:
"@angular/core": "~7.2.0",
"highcharts": "7.0.1",
"highcharts-angular": "~2.4.0",

**My HTML**: 
```
<highcharts-chart 
    [Highcharts]="Highcharts"

    [constructorType]="chartConstructor"
    [options]="chartOptions"
    [callbackFunction]="chartCallback"

    [(update)]="updateFlag"
    [oneToOne]="oneToOneFlag"
    [runOutsideAngular]="runOutsideAngularFlag"

    class="chart"
  ></highcharts-chart>
```

**My Chart options in TypeScript component file:**
```
this.chartOptions = {
      chart: {
        panning: true,
        spacingLeft: 0,
        spacingBottom: 0,
        // spacingLeft: 50,
        // alignTicks: false,
        plotBackgroundColor: 'transparent',
        events: {
          load: function () {
            console.log('load event called');
            const fullScreenButton = this.elementRef.nativeElement.querySelector('.highcharts-full-screen');
            // this.Highcharts.removeEvent();

            this.Highcharts.addEvent(
              fullScreenButton,
              'click',
              () => {
                console.log('User has clicked on "full-screen" button');
              }
            );
          }
        }
      },

      title: {
        text: '' // Empty title = no title
      },

      legend: {
        enabled: false
      },

      // Define yAxis settings
      yAxis: [
        {
          labels: {
            align: 'left'
          },
          height: '70%',
          resize: {
            enabled: true
          }
        },
        {
          top: '60%',
          height: '10%',
          offset: 0
        },
        {
          labels: {
            align: 'left'
          },
          top: '75%',
          height: '25%',
          offset: 0
        }
      ],

      // Handle top-left custom tooltip
      tooltip: {
        shape: 'square',
        headerShape: 'callout',
        borderWidth: 0,
        shadow: false,
        positioner: function(width, height, point) {
          const chart = this.chart;
          let position;
          if (point.isHeader) {
            position = {
              x: Math.max(
                // Left side limit
                chart.plotLeft,
                Math.min(
                  point.plotX + chart.plotLeft - width / 2,
                  // Right side limit
                  chart.chartWidth - width - chart.marginRight
                )
              ),
              y: point.plotY
            };
          } else {
            position = {
              x: point.series.chart.plotLeft,
              y: point.series.yAxis.top - chart.plotTop
            };
          }
          return position;
        }
      },

      // Customize range selector
      rangeSelector: {
        enable: false,
        allButtonsEnabled: false,
        // selected: 0,
        inputEnabled: false,
        buttonPosition: {
          align: 'left',
          x: 0,
          y: 0
        },
        inputPosition: {
          y: 0,
          align: 'right',
          x: -90
        },

      // Handle navigator at the bottom of the chart
      navigator: {
        enabled: false
      },

      // Add custom events to navigation
      navigationBindings: {
        events: {
          selectButton: function (event) {
            let newClassName = event.button.className + ' highcharts-active';
            const topButton = event.button.parentNode.parentNode;

            if (topButton.classList.contains('right')) {
              newClassName += ' right';
            }

            // If this is a button with sub buttons,
            // change main icon to the current one:
            if (!topButton.classList.contains('highcharts-menu-wrapper')) {
              topButton.className = newClassName;
            }

            // Store info about active button:
            this.chart.activeButton = event.button;
          },
          deselectButton: function (event) {
            event.button.parentNode.parentNode.classList.remove('highcharts-active');

            // Remove info about active button:
            this.chart.activeButton = null;
          },
          showPopup: function (event) {
            if (!this.indicatorsPopupContainer) {
              this.indicatorsPopupContainer = document
                  .getElementsByClassName('highcharts-popup-indicators')[0];
            }

            if (!this.annotationsPopupContainer) {
              this.annotationsPopupContainer = document
                .getElementsByClassName('highcharts-popup-annotations')[0];
            }

            if (event.formType === 'indicators') {
              this.indicatorsPopupContainer.style.display = 'block';
            } else if (event.formType === 'annotation-toolbar') {
              // If user is still adding an annotation, don't show popup:
              if (!this.chart.activeButton) {
                this.chart.currentAnnotation = event.annotation;
                this.annotationsPopupContainer.style.display = 'block';
              }
            }
          },
          closePopup: function () {
            this.indicatorsPopupContainer.style.display = 'none';
            this.annotationsPopupContainer.style.display = 'none';
          }
        }
      },

      // Enable stock tools
      stockTools: {
        gui: {
          enabled: true
        }
      },

      // Define chart series
      series: [
        {
          type: 'candlestick',
          id: 'serie1',
          name: this.activeInstrumentTicker$,
          data: this.chartData,
          yAxis: 0,
          zIndex: 1,
        },
        {
          type: 'column',
          id: 'volume',
          name: 'Volume',
          data: this.volumeData,
          yAxis: 2
        },
      ],

      // Handle responsive mode
      responsive: {
        rules: [
          {
            condition: {
              maxWidth: 800
            },
            chartOptions: {
              rangeSelector: {
                inputEnabled: false
              }
            }
          }
        ]
      }

    };
```

Reading the official API Doc I was expecting to see a console log of "'load event called" after chart init, but nothing happends. 

Is that an "highcharts-angular" error or am I making some mistake?

Thanks

**Comment by KacperMadej** [comment]
Hi @frabanca 

Thank you for contacting us about the problem.

I'm unable to recreate the problem - the load event works fine: https://codesandbox.io/s/3q75k421m5

Maybe there are some missing modules that must be loaded?
Please try to update the live demo to show the issue that you have. A minimal demo would be nice, but any showing the problem will be enough.

**Comment by frabanca** [comment]
Hi @KacperMadej 

Thank you for your quick reply and for the codesendbox. 
I've tryed to replicate bug but I can't. It turns out the problem was (my bad!) a wrong initializations of the chartOptions property. 

I was initializating that property like that: chartOptions = {} and than I was adding the options after ngOnInit. To make it work I had to init the chartOptions property as a type --> chartOptions: {} <--

**Wrong Code:** 
```
chartOptions = {};

ngOnInit() {
  this.initChart();
}

initChart() {
  chartOptions = {
    ...
  }
}
```

**Good Code:**
```
chartOptions: {};

ngOnInit() {
  this.initChart();
}

initChart() {
  chartOptions = {
    ...
  }
}
```

I'v done a codesendbox too to better explain what happened: https://codesandbox.io/s/239m39my30

Thanks again!
